### PR TITLE
chore: convert tests to TS

### PR DESCRIPTION
**Pull request by vladfrangu**

### CODE REVIEW

**Review by B4nan — CHANGES_REQUESTED** [review]
looking good, i'd just keep the autoscaled pool properties as private, plus added few nitpicks

File: test/autoscaling/snapshotter.test.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ describe('Snapshotter', () => {
     test('should collect snapshots with some values', async () => {
         // mock client data
         const oldStats = utils.apifyClient.stats;
-        utils.apifyClient.stats = {};
+        utils.apifyClient.stats = {} as never;
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply test/autoscaling/snapshotter.test.ts:32]: Yeah, casting as never seeems to have the same effect as casting as any. It's mostly for simplicity sake, an @ts-expect-error would've been sufficient too

File: test/utils.test.ts [hunk 2]
```diff
@@ -46,56 +47,78 @@ describe('utils.addCharsetToContentType()', () => {
...
-            .isDocker(true)
-            .then((is) => {
-                expect(is).toBe(true);
-                fs.stat.restore();
-                fs.readFile.restore();
-            });
+    test('works for dockerenv && cgroup', async () => {
+        const statStub = sinon
+            .stub(fs, 'stat')
+            // @ts-expect-error sinon doesn't pick up certain FS overloads
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply test/utils.test.ts:53]: The mocking works, the types don't quite work due to overloads 😓

**Review by B4nan — APPROVED** [review]
any reason this is a draft PR or can we merge it?

**Comment by vladfrangu** [comment]
Yeah it's draft cause I'm working on all tests in it. I can split them up I suppose, if that's better for you 👍

**Comment by B4nan** [comment]
lets merge it now as the code is reviewed. you can then reuse the same branch and open new PR with new changes, right?

**Comment by vladfrangu** [comment]
Yep, I can. Let's do it that way then

**Comment by B4nan** [comment]
One test is failing and looks like it fails all the time, probably some mocks dont work properly?

**Comment by vladfrangu** [comment]
Nop, me fixing a type error in one of the tests caused it to always error, so I just pushed a fix to the branch but you merged the PR before it was pushed haha. Just merge the branch again and tests should pass again

**Pull request merged**